wordpress-mobile/WordPress-Android
# Add reblog to Reader Post Details

Fixes #11531 

## Description

This feature adds the reblog functionality in the reader post details view

A reblog button has been added in the Reader Post Details. 
If the user has more that one sites the button triggers the existing Site Picker.

| Post Details |Site Picker (unchanged)|
|--- |---|
|![2020-03-19 22 05 24](https://user-images.githubusercontent.com/304044/77110680-993a5c00-6a2e-11ea-8caf-e9f7b49b9903.png)|![2020-03-19 22 05 33](https://user-images.githubusercontent.com/304044/77110683-9b041f80-6a2e-11ea-8f21-198d89d5ead1.png)|

If the user has only one site the editor is triggered directly.
At the moment handling users with no site is not supported.

When the users selects a site he is redirected to the editor with the following information prefilled:
* Title (with source)
* Image (if the post has a featured)
* Quote
* Citation

| Gutenberg editor with featured image| Gutenberg editor without image |
| --- | --- |
|![device-2020-03-22-114527](https://user-images.githubusercontent.com/304044/77246840-b8560b00-6c33-11ea-8b38-28dc15e7b888.png)|![2020-03-19 22 06 05](https://user-images.githubusercontent.com/304044/77110686-9c354c80-6a2e-11ea-858d-f09c597f6348.png)|

| Aztec editor with featured image| Aztec editor without image |
| --- | --- |
|![device-2020-03-22-114630](https://user-images.githubusercontent.com/304044/77246853-d459ac80-6c33-11ea-9059-40c0c62eb03f.png)|![device-2020-03-22-114609](https://user-images.githubusercontent.com/304044/77246855-d885ca00-6c33-11ea-8bea-9943adfebb79.png)|

When the user taps the Publish button and confirms he returns to the original context

| Uploading State| Post Published |
| --- | --- |
|![device-2020-03-25-220432](https://user-images.githubusercontent.com/304044/77580504-d1391780-6ee4-11ea-8eba-eeaf05132dad.png)|![device-2020-03-25-220329](https://user-images.githubusercontent.com/304044/77580509-d4cc9e80-6ee4-11ea-9bb5-455d412b5209.png)|

## Test

0. Open the App and Login if needed
1. Select the Reader tab
2. Tap on a post card
3. Notice the reblog action on the post detail view
4. Tap on the reblog action
5. If prompted select a site from the picker
6. Notice that the editor opens with content from the original post: Title (with source), Image (if the post has a featured), Quote and Citation link
7. Optionally edit the post
8. Press the Publish button
9. Confirm by pressing the Publish Now button in the dialog
10. Notice that you return to the original context (post detail view)
11. Notice the snackbar with the VIEW button
12. Tap on the VIEW button
13. Examine the reblogged post in the browser

PR submission checklist:
